miio/heater.py

```diff
@@ -0,0 +1,279 @@
+                values_count,
+            )
+
+        return HeaterStatus(dict(zip(properties, values)))
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_power", ["on"])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    @command(
+        click.argument("temperature", type=int),
+        default_output=format_output("Setting target temperature to {temperature}"),
+    )
+    def set_target_temperature(self, temperature: int):
+        """Set target temperature."""
+        if self.model == MODEL_HEATER_ZA1 and (temperature < 16 or temperature > 32):
+            raise HeaterException("Invalid target temperature: %s" % temperature)
+        if self.model == MODEL_HEATER_MA1 and (temperature < 20 or temperature > 32):
+            raise HeaterException("Invalid target temperature: %s" % temperature)
+
+        return self.send("set_target_temperature", [temperature])
+
+    @command(
+        click.argument("brightness", type=EnumType(Brightness, False)),
+        default_output=format_output("Setting display brightness to {brightness}"),
+    )
+    def set_brightness(self, brightness: Brightness):
+        """Set display brightness."""
+        return self.send("set_brightness", [brightness.value])
+
+    @command(
+        click.argument("buzzer", type=bool),
+        default_output=format_output(
+            lambda buzzer: "Turning on buzzer" if buzzer else "Turning off buzzer"
+        ),
+    )
+    def set_buzzer(self, buzzer: bool):
+        """Set buzzer on/off."""
+        if buzzer:
+            return self.send("set_buzzer", ["on"])
+        else:
+            return self.send("set_buzzer", ["off"])
+
+    @command(
+        click.argument("lock", type=bool),
+        default_output=format_output(
+            lambda lock: "Turning on child lock" if lock else "Turning off child lock"
+        ),
+    )
+    def set_child_lock(self, lock: bool):
+        """Set child lock on/off."""
+        if lock:
+            return self.send("set_child_lock", ["on"])
+        else:
+            return self.send("set_child_lock", ["off"])
+
+    @command(
+        click.argument("seconds", type=int),
+        default_output=format_output("Setting delayed turn off to {seconds} seconds"),
+    )
+    def delay_off(self, seconds: int):
+        """Set delay off seconds."""
+        if self.model == MODEL_HEATER_ZA1:
+            if seconds < 0 or seconds > 9 * 3600:
+                raise HeaterException("Invalid delay time: %s" % seconds)
+            return self.send("set_poweroff_time", [seconds])
+
+        elif self.model == MODEL_HEATER_MA1:
+            if seconds < 0 or seconds > 5 * 3600:
+                raise HeaterException("Invalid delay time: %s" % seconds)
+            return self.send("set_poweroff_level", [seconds // 3600])
```

Comment:
MA1 accepts the values only in hours? What's the behavior of setting these to zero (<3600 for MA1)?

Reply:
I have personal access only to za1 model. For ma1 I have only tcpdump, token and brief list of what was done in mihome app, so I can't confirm the behavior.

Reply:
Fair enough :+1: